### ISSUE TITLE
Using sizeColumnsToFit Vertical Scrollbar overlaps column hence horizontal scrollbar appears

### ISSUE THREAD

**Issue by Lazarencjusz** [ag-grid/ag-grid#2310]
**I'm submitting a ...**
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
I am using `sizeColumnsToFit` to fit grid width to wrapper container and horizontal scrollbar appears because of vertical scrollbar overlaps column instead of to be placed like in new column (see screenshot):
![image](https://user-images.githubusercontent.com/6573466/38737888-f3d05302-3f30-11e8-97f7-634a290ea8a5.png)

**Expected behavior**
Vertical scrollbar should be placed like in new column and be the new column with scrollbar should be taken to recalculation of column width like in other case in my app:
![image](https://user-images.githubusercontent.com/6573466/38738173-b673a18e-3f31-11e8-859c-81e786780008.png)

When I remove app container scrollbar it works fine, however, I can't do it on production. When there is less rows, vertical scrollbar not appears, and width is ok. I have no idea what cause this behavior. The grid options are same both cases:
````
{
        onGridReady: this.onGridReady,
        columnDefs: this.props.columns,
        context: { componentParent: this },
        enableColResize: false,
        frameworkComponents: {
         /*...*/
        },
        onRowSelected: this.handleRowSelected,
        rowBuffer: 5,
        rowData: this.props.dataProvider,
        rowDeselection: false,
        rowSelection: 'single',
        suppressCellSelection: true,
        suppressContextMenu: true,
        suppressCopyRowsToClipboard: true,
        suppressDragLeaveHidesColumns: true,
        suppressMenuColumnPanel: true,
        suppressMenuFilterPanel: true,
        suppressMenuMainPanel: true,
        suppressNoRowsOverlay: true,
        width: this.props.width,
        height: this.height
      },
````

**Minimal reproduction of the problem with instructions**
At the begging please advise what could be a problem

**What is the motivation / use case for changing the behavior?**
Inconsistency.

**Please tell us about your environment:**
React@15.x

* **ag-Grid version:**
14.0.1

* **Browser:**
Chrome v.63 IE v.11

* **Language:** [ES6]

**Comment by dredwolff** [comment]
I am also running into this issue, while the workaround Lazarencjusz uses may work for him, in angular the equivalent call would be ngAfterViewChecked() which gets called a *lot*.

Even the ag-grid example: https://www.ag-grid.com/javascript-grid-resizing/#sizing-columns-by-default has the same problem in all browsers

**Comment by makinggoodsoftware** [comment]
Hi,

In ag-grid we have recently started processing github issues as well as customer issues. We are also going through the backlog of issues already open and acting on them as necessary, on this particular case, it seems like this issue could be gone in our latest release.

Could you update the ag-grid version in your end and try again?

Note that this issue is getting closed, but you are welcomed to create a new issue if you think this issue is still not resolved

Thanks

**Comment by chris-rodwell** [comment]
Should anyone else have a problem with this you can workaround it by binding to the  modelUpdated event and calling api.sizeColumnsToFit() in there instead

**Comment by bnohad** [comment]
> Should anyone else have a problem with this you can workaround it by binding to the modelUpdated event and calling api.sizeColumnsToFit() in there instead

very bad hack don't do it, the `modelUpdate` is being called very often

**Comment by bnohad** [comment]
still having this bug in version `^25.2.0` any updates?

**Comment by bdwade100** [comment]
I added the following CSS and it seemed to fix the problem (I'm using ag-grid 28.1.1):

```
.ag-center-cols-viewport {
    overflow-x: hidden;
}
```
